plans/network/pingpong.go

```diff
@@ -187,6 +204,92 @@ func pingpong(runenv *runtime.RunEnv, initCtx *run.InitContext) error {
 	return nil
 }
 
+// Exchanges addresses with other instances in the test, and returns their addresses as a slice
+func exchangeListenAddrs(ctx context.Context, runenv *runtime.RunEnv, client sync.Client) (*ListenAddrs, error) {
+
+	tcpAddr, err := getSubnetAddr(runenv.TestSubnet)
+	if err != nil {
+		return nil, err
+	}
+
+	mynode := &ListenAddrs{}
+	mine := map[string]struct{}{}
+
+	mynode.Addrs = append(mynode.Addrs, tcpAddr.String())
+	for _, l := range mynode.Addrs {
+		mine[l] = struct{}{}
+	}
+
+	_ = client.MustSignalAndWait(ctx, sync.State("listening"), runenv.TestInstanceCount)
+
+	allAddrs, err := shareAddresses(ctx, client, runenv, mynode)
+	if err != nil {
+		return nil, err
+	}
+
+	otherAddrs := []string{}
+	// filter addresses to remove any own address
+	for _, addr := range allAddrs {
+		if _, ok := mine[addr]; ok {
+			continue
+		}
+		otherAddrs = append(otherAddrs, addr)
+	}
+
+	_ = client.MustSignalAndWait(ctx, sync.State("got-other-addrs"), runenv.TestInstanceCount)
+
+	retVal := &ListenAddrs{}
+	retVal.Addrs = otherAddrs
+	return retVal, nil
+}
+
+// Returns the IP address belonging to the given subnet
+func getSubnetAddr(subnet *ptypes.IPNet) (*net.TCPAddr, error) {
+	addrs, err := net.InterfaceAddrs()
+	if err != nil {
+		return nil, err
+	}
+	for _, addr := range addrs {
+		if ip, ok := addr.(*net.IPNet); ok {
+			if subnet.Contains(ip.IP) {
+				tcpAddr := &net.TCPAddr{IP: ip.IP}
+				return tcpAddr, nil
+			} else {
+				fmt.Printf("%s does not contain %s\n", subnet, ip.IP)
+			}
+		} else {
+			panic(fmt.Sprintf("%T", addr))
+		}
+	}
+	return nil, fmt.Errorf("no network interface found. Addrs: %v", addrs)
+}
+
+// Shares all of this instance's addresses with all other instances in the test
```

Comment:
I don't think I quite understand this. What other addresses than https://github.com/testground/testground/blob/330ca290d6aa44d06b0b39268744910e35924871/plans/network/pingpong.go#L210 are we sharing?

Reply:
Nice catch - I'll rewrite this so that only a single address (data IP) gets shared.

Reply:
Hey, thanks for making the updates. Sorry I wasn't clearer on what I meant - I had something like this in mind: https://github.com/testground/testground/pull/1470 (feel free to merge if you like it; I targeted your PR there) - i.e. taking advantage of the fact that we only ever need to publish a single string, not a list of address structs.

Reply:
Oh, now I understand - thanks for making the PR, I merged it!